src/languages/clojure.js

```diff
@@ -8,8 +8,8 @@ Category: lisp
 
 /** @type LanguageFn */
 export default function(hljs) {
-  const SYMBOLSTART = 'a-zA-Z_\\-!.?+*=<>&#\'';
-  const SYMBOL_RE = '[' + SYMBOLSTART + '][' + SYMBOLSTART + '0-9/;:]*';
+  const SYMBOLSTART = 'a-zA-Z_\\-!.?+*=<>&\'';
```

Comment:
`#` meant to be removed?

Reply:
I readded it in line 12 because it interfered with my fix for namespaced maps.

```js
const SYMBOL_RE = '[#]?[' + SYMBOLSTART + '][' + SYMBOLSTART + '0-9/;:$#]*';
```

If a symbol begins with `#` it can't be followed by a number, and it can't stand alone.

`#::{}`, `#::foo{}`, `#:foo{}` are valid maps. `#::` shouldn't be matched as a symbol, but the start of a namespaced map. 


The "SYMBOL" mode matches way more than the Clojure term "symbol", maybe we should split it into two? 
symbols and specials? 

> **Symbols**
Symbols begin with a non-numeric character and can contain alphanumeric characters and *, +, !, -, _, ', ?, <, > and = (other characters may be allowed eventually).
>
> '/' has special meaning, it can be used once in the middle of a symbol to separate the namespace from the name, e.g. my-namespace/foo. '/' by itself names the division function.
>
>'.' has special meaning - it can be used one or more times in the middle of a symbol to designate a fully-qualified class name, e.g. java.util.BitSet, or in namespace names. Symbols beginning or ending with '.' are reserved by Clojure. Symbols containing / or . are said to be 'qualified'.
>
> Symbols beginning or ending with ':' are reserved by Clojure. A symbol can contain one or more non-repeating ':'s.

https://clojure.org/reference/reader#_symbols

Reply:
This PR is big enough but if you want to circle back later with more improvements. :)